salt/runners/jobs.py

```diff
@@ -305,68 +305,78 @@ def list_jobs(ext_source=None,
         )
     mminion = salt.minion.MasterMinion(__opts__)
 
-    ret = mminion.returners['{0}.get_jids'.format(returner)]()
-
-    mret = {}
-    for item in ret:
-        _match = True
-        if search_metadata:
-            _match = False
-            if 'Metadata' in ret[item]:
-                if isinstance(search_metadata, dict):
-                    for key in search_metadata:
-                        if key in ret[item]['Metadata']:
-                            if ret[item]['Metadata'][key] == search_metadata[key]:
+    if returner in [ 'pgjsonb']:
```

Comment:
Coupling the type of returner in use to the jobs runner itself seems undesirable to me.

Reply:
Today the jobs API is unusable for a real production, and no one download a database to filter it in his code, so I'm agree it's not very nice to have this hardcoded here but I don't find another easy solution. Have you got an idea ?

Reply:
Hmm. Could we mark certain returners as accepting a job_filter, perhaps via a configuration option? I'm totally in favor of the idea overall, but I want to make sure it's something we can support broadly and not just as a one-off.

Reply:
Also, I don't understand why we don't store the job result like this and not how we want it when calling it again ?